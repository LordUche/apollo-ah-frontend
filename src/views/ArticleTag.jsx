import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
/**
 * @returns {JSX} - The JSX representation of the article's tags
 *
 */
function Tag({ tag }) {
  return (
    <Link to={`/search?tag=${tag.tagName}`} className="badge">
      {tag.tagName}
    </Link>
  );
}
Tag.propTypes = {
  tag: PropTypes.object.isRequired,
};

export default Tag;
