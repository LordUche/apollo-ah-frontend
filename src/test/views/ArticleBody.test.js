import React from 'react';
import { mount } from 'enzyme';
import ArticleTag from '../../views/ArticleBody';
import setup from '../setup';

const props = {
  bookmarkArticle: jest.fn(),
  bookmarked: false,
  article: {
    title: 'title',
    description: 'description',
    body: 'body',
    image: 'image.jpg',
    tagList: []
  }
};

describe('<ArticleBody>', () => {
  it('should render without crashing when there are no tags', () => {
    const wrapper = mount(
      <ArticleTag
        bookmarkArticle={props.bookmarkArticle}
        bookmarked={props.bookmarked}
        article={props.article}
        />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing when there are tags', () => {
    props.article.tagList = [
      {
        tagName: 'tech',
      }
    ];
    const wrapper = setup(
      <ArticleTag
        bookmarkArticle={props.bookmarkArticle}
        bookmarked={props.bookmarked}
        article={props.article}
        />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
