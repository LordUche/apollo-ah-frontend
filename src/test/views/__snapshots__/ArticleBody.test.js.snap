// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleBody> should render without crashing when there are no tags 1`] = `
<ArticleBody
  article={
    Object {
      "body": "body",
      "description": "description",
      "image": "image.jpg",
      "tagList": Array [],
      "title": "title",
    }
  }
  bookmarkArticle={[MockFunction]}
  bookmarked={false}
>
  <div>
    <h1
      className="signle-title"
    >
      title
    </h1>
    <p
      className="single-sub"
      style={
        Object {
          "color": "#9c9c9c",
          "fontSize": 21,
          "fontWeight": "bold",
        }
      }
    >
      description
       
    </p>
    <div
      className="single__author__container"
    >
      <div
        className="article__author"
      >
        <div
          className="author__profile--mini"
        >
          <img
            alt=""
            src="image.jpg"
          />
          <div
            className="author__profile--details"
          >
            <h6>
              Alexandra
            </h6>
            <span>
              22 hrs ago
            </span>
            <span
              className="article-author-dot"
            />
            <span>
              4 min read
            </span>
          </div>
        </div>
        <div
          className="bookmark__div"
        >
          <button
            onClick={[MockFunction]}
            type="button"
          >
            <span
              className="fas fa-bookmark transition "
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="single-body"
    >
      body
    </div>
    <div
      className="single-tags text-center"
    />
    <div
      className="pg-empty-placeholder"
    />
    <div
      className="clap-grp"
    >
      <span
        className="clap-icon"
      >
        <img
          alt=""
          src={Object {}}
        />
        40k claps
      </span>
      <div
        className="share-grp"
      >
        <a
          className="text-muted"
          href="you"
        >
          <i
            className="fas fa-bookmark"
          />
        </a>
        <a
          className="text-muted"
          href="you"
        >
          <i
            className="fas fa-ellipsis-h"
          />
        </a>
      </div>
      <div
        className="share-grp"
      >
        <SocialShare>
          <div>
            <a
              className="text-danger"
              href="https://mail.google.com/mail/u/0/?view=cm&fs=1&to&su=Authors+Haven&body=http://localhost/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                className="fas fa-envelope"
              />
            </a>
            <a
              className="text-primary"
              href="https://www.facebook.com/sharer.php?u=http://localhost/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                className="fab fa-facebook-square"
              />
            </a>
            <a
              className="text-white"
              href="https://twitter.com/intent/tweet?text=http://localhost/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <i
                className="fab fa-twitter-square text-info"
              />
            </a>
          </div>
        </SocialShare>
      </div>
    </div>
    <div
      className="single-author-container single-article-author"
    >
      <div
        className="article-author"
      >
        <img
          alt=""
          src="image.jpg"
        />
        <div>
          <h6>
            Alexandra
          </h6>
          <p>
            Journalist, author, dad. Former TIME magazine editor and foreign correspondent. 
          </p>
          <button
            className="btn btn-light"
            type="button"
          >
            Follow
          </button>
        </div>
      </div>
    </div>
  </div>
</ArticleBody>
`;

exports[`<ArticleBody> should render without crashing when there are tags 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ArticleBody
        article={
          Object {
            "body": "body",
            "description": "description",
            "image": "image.jpg",
            "tagList": Array [
              Object {
                "tagName": "tech",
              },
            ],
            "title": "title",
          }
        }
        bookmarkArticle={[MockFunction]}
        bookmarked={false}
      >
        <div>
          <h1
            className="signle-title"
          >
            title
          </h1>
          <p
            className="single-sub"
            style={
              Object {
                "color": "#9c9c9c",
                "fontSize": 21,
                "fontWeight": "bold",
              }
            }
          >
            description
             
          </p>
          <div
            className="single__author__container"
          >
            <div
              className="article__author"
            >
              <div
                className="author__profile--mini"
              >
                <img
                  alt=""
                  src="image.jpg"
                />
                <div
                  className="author__profile--details"
                >
                  <h6>
                    Alexandra
                  </h6>
                  <span>
                    22 hrs ago
                  </span>
                  <span
                    className="article-author-dot"
                  />
                  <span>
                    4 min read
                  </span>
                </div>
              </div>
              <div
                className="bookmark__div"
              >
                <button
                  onClick={[MockFunction]}
                  type="button"
                >
                  <span
                    className="fas fa-bookmark transition "
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            className="single-body"
          >
            body
          </div>
          <div
            className="single-tags text-center"
          >
            <Tag
              key="tech"
              tag={
                Object {
                  "tagName": "tech",
                }
              }
            >
              <Link
                className="badge"
                replace={false}
                to="/search?tag=tech"
              >
                <a
                  className="badge"
                  href="/search?tag=tech"
                  onClick={[Function]}
                >
                  tech
                </a>
              </Link>
            </Tag>
          </div>
          <div
            className="pg-empty-placeholder"
          />
          <div
            className="clap-grp"
          >
            <span
              className="clap-icon"
            >
              <img
                alt=""
                src={Object {}}
              />
              40k claps
            </span>
            <div
              className="share-grp"
            >
              <a
                className="text-muted"
                href="you"
              >
                <i
                  className="fas fa-bookmark"
                />
              </a>
              <a
                className="text-muted"
                href="you"
              >
                <i
                  className="fas fa-ellipsis-h"
                />
              </a>
            </div>
            <div
              className="share-grp"
            >
              <SocialShare>
                <div>
                  <a
                    className="text-danger"
                    href="https://mail.google.com/mail/u/0/?view=cm&fs=1&to&su=Authors+Haven&body=http://localhost/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <i
                      className="fas fa-envelope"
                    />
                  </a>
                  <a
                    className="text-primary"
                    href="https://www.facebook.com/sharer.php?u=http://localhost/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <i
                      className="fab fa-facebook-square"
                    />
                  </a>
                  <a
                    className="text-white"
                    href="https://twitter.com/intent/tweet?text=http://localhost/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    <i
                      className="fab fa-twitter-square text-info"
                    />
                  </a>
                </div>
              </SocialShare>
            </div>
          </div>
          <div
            className="single-author-container single-article-author"
          >
            <div
              className="article-author"
            >
              <img
                alt=""
                src="image.jpg"
              />
              <div>
                <h6>
                  Alexandra
                </h6>
                <p>
                  Journalist, author, dad. Former TIME magazine editor and foreign correspondent. 
                </p>
                <button
                  className="btn btn-light"
                  type="button"
                >
                  Follow
                </button>
              </div>
            </div>
          </div>
        </div>
      </ArticleBody>
    </Router>
  </BrowserRouter>
</Provider>
`;
